src/bp/core/routers/bots/content.ts

```diff
@@ -129,6 +134,70 @@ export class ContentRouter extends CustomRouter {
         res.sendStatus(200)
       })
     )
+
+    this.router.get(
+      '/export',
+      this._checkTokenHeader,
+      this._needPermissions('read', 'bot.content'),
+      async (req, res) => {
+        const elements = await this.cms.getAllElements(req.params.botId)
+        const filtered = elements.map(x => _.omit(x, ['createdBy', 'createdOn', 'modifiedOn']))
+
+        res.setHeader('Content-Type', 'application/json')
+        res.setHeader('Content-disposition', `attachment; filename=content_${moment().format('DD-MM-YYYY')}.json`)
+        res.end(JSON.stringify(filtered, undefined, 2))
+      }
+    )
+
+    const upload = multer()
+    this.router.post(
+      '/analyzeImport',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      upload.single('file'),
+      this.asyncMiddleware(async (req: any, res) => {
+        try {
+          const existingElements = await this.cms.getAllElements(req.params.botId)
+          const contentTypes = (await this.cms.getAllContentTypes(req.params.botId)).map(x => x.id)
+
+          const importData = (await validate(JSON.parse(req.file.buffer), CmsImportSchema)) as ContentElement[]
+          const importedContentTypes = _.uniq(importData.map(x => x.contentType))
+
+          res.send({
+            cmsCount: (existingElements && existingElements.length) || 0,
+            fileCmsCount: (importData && importData.length) || 0,
+            missingContentTypes: _.difference(importedContentTypes, contentTypes)
+          })
+        } catch (err) {
+          throw new InvalidOperationError(`Error importing your file: ${err}`)
+        }
+      })
+    )
+
+    this.router.post(
+      '/import',
+      this._checkTokenHeader,
+      this._needPermissions('write', 'bot.content'),
+      upload.single('file'),
+      async (req: any, res) => {
+        if (req.body.action === 'clear_insert') {
+          await this.cms.deleteAllElements(req.params.botId)
+        }
+
+        try {
+          const importData: ContentElement[] = await JSON.parse(req.file.buffer)
+
+          for (const { contentType, formData, id } of importData) {
+            await this.cms.createOrUpdateContentElement(req.params.botId, contentType, formData, id)
```

Comment:
even if not cleaar_insert, this will overwrite all elements ? 

is clear_insert only to delete missing content elements in inport data ?

Reply:
Yep, the message displayed to the user is more clear when making a choice. You can either create new elements and update existing ones, or clear all of the existing ones, then create them